Event/event.js

```diff
@@ -1,17 +1,81 @@
 var eventList = [];
+var FreeEvents = [];
+var BusyEvents = [];
+const NEXT_WEEK=7*24*60*60*1000; //same time next week in mill sec
+const NEXT_SLOT=30*60*1000; //next 30 min in mill sec
 
-var Event = function(opening, recurring, startDate, endDate){
+var Event = function (opening, recurring, startDate, endDate) {
     this.opening = opening;
     this.recurring = recurring;
     this.startDate = startDate;
     this.endDate = endDate;
-
     eventList.push(this);
 };
 
-Event.prototype.availabilities = function(fromDate, toDate){
-    return //Something awesome;
+Event.prototype.availabilities = function (fromDate, toDate) {
+    eventList.forEach(function (element) {
+        if (element.opening === true) {
+            //  get free slot (start & end) in mill sec
+            start_open = Date.parse(element.startDate);
+            end_open = Date.parse(element.endDate);
+            AddFreeEvents(element, Date.parse(fromDate), Date.parse(toDate), start_open, end_open);
+        };
+    });
+    eventList.forEach(function (element) {
+        if (element.opening === false) {
+            //  get busy windows (start & end) in mill sec
+            start_busy = Date.parse(element.startDate);
+            end_busy = Date.parse(element.endDate);
+            AddBusyEvents(element, Date.parse(fromDate), Date.parse(toDate), start_busy, end_busy);
+        };
+    });
+    //  available windows = free windows - busy windows
+    let availabilities = FreeEvents.filter(element => !BusyEvents.includes(element));
+
+    //print availabilities
+    if (availabilities.length == 0) {
+        console.log('I\'m not available at all.');
+    } else {
+        availabilities.forEach(function (slot) {
+            console.log('I\'m available the ' + slot);
+        });
+        console.log('I\'m not available any other time !');
+    }
 };
 
-module.exports = Event;
+function AddFreeEvents(element, fromDate, toDate, start_open, end_open) {
+    start_date=Math.max(fromDate,start_open);
+    end_date=Math.min(toDate,end_open);
+    next_slot = start_date;
+    while (next_slot < end_date){
+        FreeEvents.push(timeToString(next_slot));
+        next_slot += NEXT_SLOT; // check next 30 min slot
+    };
+    //  recursive call for recurring available events
+    if (element.recurring === true && start_open < toDate) {
```

Comment:
I don't think we need to compare to true here

Reply:
> It would have been interesting to write one or 2 tests

I wrote a test case in the main.js (without adding the results)